### PR TITLE
bug(condition) Adding langauge to product has status condition

### PR DESCRIPTION
**Pull request by wiewiurdp**
# Description

 Adding langauge to product has status condition

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [x] Unit Tests
- [x] Behat Test

# Checklist:
- [x] I have read the contribution requirements and fulfill them.
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by wfajczyk** [comment]
You don't added migration for currently stored conditions.

File: module/condition/migrations/Version20201119103017.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+class Version20201119103017 extends AbstractErgonodeMigration
+{
+
+    public function up(Schema $schema): void
+    {
+        $languages = $this->getDictionaryActive();
+
+        $this->addSql(sprintf(
+            'WITH conditions_list as (
+                SELECT id, (\'{\'|| arr.position -1 || \',language}\')::text[] as path
```
Comment (piotrkreft) [inline module/condition/migrations/Version20201119103017.php:23]: Is there any specific reason for not extracting just position and creating this in `jsonb_set` function? I'd rather have it like that

File: module/condition/migrations/Version20201119103017.php [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                FROM condition_set,
+                    jsonb_array_elements(conditions) with ordinality arr(item_object, position)
+                where arr.item_object::jsonb ->> \'type\' = \'PRODUCT_HAS_STATUS_CONDITION\')
+            UPDATE condition_set
+            set conditions = jsonb_set(conditions, conditions_list.path, \'%s\')
```
Comment (piotrkreft) [inline module/condition/migrations/Version20201119103017.php:28]: Please write language constructs (like `set`)  with the upper case in the entire query and not just parts of it. Also reformat it - i.e. in this line indentation is missing

File: module/condition/migrations/Version20201119103017.php [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            from conditions_list
+            where condition_set.id = conditions_list.id',
+            $languages
+        ));
+
+        $this->addSql(sprintf(
+            'WITH conditions_list as (
```
Comment (piotrkreft) [inline module/condition/migrations/Version20201119103017.php:35]: Here as in above.

File: module/condition/migrations/Version20201119103017.php [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+            $languages
+        ));
+    }
+
+    /**
+     * @return array
+     */
+    public function getDictionaryActive(): string
+    {
+        $qb = $this->connection->createQueryBuilder()
```
Comment (piotrkreft) [inline module/condition/migrations/Version20201119103017.php:55]: In fact, you don't need the connection service to execute the query. Just add the appropriate SQL to the above queries.

File: module/condition/src/Infrastructure/Condition/Calculator/ProductHasStatusConditionCalculatorStrategy.php [hunk 5]
```diff
@@ -45,11 +46,14 @@ public function calculate(AbstractProduct $product, ConditionInterface $configur
...
         Assert::true($product->hasAttribute($statusAttributeCode));
-        $value = $product->getAttribute($statusAttributeCode)->getValue();
-        $productStatusId = new StatusId(reset($value));
+        $productStatuses = $product->getAttribute($statusAttributeCode)->getValue();
+        $filteredProductStatuses = $this->getStatusIdsByLanguages($configuration->getLanguage(), $productStatuses);
+
         $result = [];
-        foreach ($configuration->getValue() as $searchedStatusId) {
-            $result[] = $productStatusId->isEqual($searchedStatusId);
+        foreach ($filteredProductStatuses as $filteredProductStatus) {
```
Comment (piotrkreft) [inline module/condition/src/Infrastructure/Condition/Calculator/ProductHasStatusConditionCalculatorStrategy.php:53]: Multiple iterations over the same array occur - maybe worth reducing?

**Review by piotrkreft — CHANGES_REQUESTED** [review]

**Review by rprzedzik — DISMISSED** [review]

**Review by wfajczyk — APPROVED** [review]

**Review by wfajczyk — APPROVED** [review]

**Review by piotrkreft — APPROVED** [review]

**Pull request merged**